lib/ddtrace/correlation.rb

```diff
@@ -1,32 +1,34 @@
+require 'ddtrace/ext/correlation'
 require 'ddtrace/environment'
 
 module Datadog
   # Contains behavior for managing correlations with tracing
   # e.g. Retrieve a correlation to the current trace for logging, etc.
   module Correlation
     # Struct representing correlation
-    Identifier = Struct.new(:trace_id, :span_id, :env) do
+    Identifier = Struct.new(:trace_id, :span_id, :env, :version) do
       def initialize(*args)
         super
         self.trace_id = trace_id || 0
         self.span_id = span_id || 0
         self.env = env || Datadog::Environment.env
+        self.version = version || Datadog::Environment.version
       end
 
       def to_s
-        str = "dd.trace_id=#{trace_id} dd.span_id=#{span_id}"
-        str += " dd.env=#{env}" unless env.nil?
+        str =  "#{Ext::Correlation::ATTR_TRACE_ID}=#{trace_id}"
+        str += " #{Ext::Correlation::ATTR_SPAN_ID}=#{span_id}"
+        str += " #{Ext::Correlation::ATTR_ENV}=#{env}" unless env.nil?
+        str += " #{Ext::Correlation::ATTR_VERSION}=#{version}" unless version.nil?
         str
       end
     end.freeze
 
-    NULL_IDENTIFIER = Identifier.new.freeze
```

Comment:
Because the `env` property of `Identifier` was being resolved at initialization time, this constant would resolve to `nil` at load time (if `DD_ENV` wasn't set) and retain this value. Later, if users configured a different `env` within the Ruby application, this instance would not use the updated value as expected, but the original value.